### DISCUSSION TITLE
Add Current Speed of the car

### DISCUSSION THREAD

**Discussion by arjanhs** [Hyundai-Kia-Connect/kia_uvo#333] [Ideas]
The information is already in the json request file but isn't used within the output. Would be nice to be added.

**Comment by cdnninja** [comment]
Are you looking for this to be part of the location or a separate sensor?  This only updates infrequently so wouldn't be very current.

**Comment by arjanhs** [comment]
Into an separate sensor so it's easy usable with HA and I'm aware it isn't accurate but gives me some insides about it.

**Comment by arjanhs** [comment]
It's in the data already now, but not yet usable as a sensor or i don't know how

NIRO EV 19 Data

**Reply by fuatakgun** [reply]
please share your attributes so we can make a sensor out of it

**Comment by arjanhs** [comment]
It's speed.value as showed below which i got from the sensor sensor.niro_ev_19_data
Extracted this myself when i was using Node-Red.

vehicle_data:
    speed:
      value: 0
      unit: 0

**Reply by fuatakgun** [reply]
to keep it up to date in home assistant, you have to call force refresh many times right? or you are fine to have it updated on very low intervals?

**Comment by arjanhs** [comment]
Yes i know and updating in slow intervals.
Also have a suggestion for a force update, when you can build a check on the 12v battery if it's on an certain level before doing the force update you can also overcome to drain the 12v battery. Used that with node-red too.
